### PR TITLE
add bash completion file

### PR DESCRIPTION
**Pull request by arthurzam**

### CODE REVIEW

**Comment by antecrescent** [comment]
The typing CI issue (https://github.com/python/typeshed/issues/11072) was fixed incorrectly upstream. I opened a new issue at https://github.com/python/typeshed/issues/12127.

To temporarily silence this issue until it is fixed upstream, you could cast `preset`:
```diff
diff --git a/pycargoebuild/__main__.py b/pycargoebuild/__main__.py
index 91a38c5..b60c702 100644
--- a/pycargoebuild/__main__.py
+++ b/pycargoebuild/__main__.py
@@ -333,7 +333,9 @@ def main(prog_name: str, *argv: str) -> int:
                                   mode="w:xz",
                                   format=tarfile.GNU_FORMAT,
                                   encoding="UTF-8",
-                                  preset=9 | lzma.PRESET_EXTREME,
+                                  preset=typing.cast(
+                                      typing.Optional[typing.Listing[0,1,2,3,4,5,6,7,8,9]],
+                                      9 | lzma.PRESET_EXTREME),
                                   ) as tar_out:  # type: ignore
                     os.fchmod(cratef.fileno(), 0o666 & ~umask)
                     logging.info("Repacking crates ...")
```

**Review by antecrescent — COMMENTED** [review]
Had a look at the code and tested it locally for a bit, LGTM. I have one question, though. Why did you add a new wheel and sdist for the previous release to the `dist` directory? I didn't find anything related to it in the code and I don't think that re-releasing 0.13.2 is what you are going for. What am I missing? :)

**Comment by arthurzam** [comment]
> Had a look at the code and tested it locally for a bit, LGTM. I have on question, though. Why did you add a new wheel and sdist for the previous release to the `dist` directory? I didn't find anything related to it in the code and I don't think that re-releasing 0.13.2 is what you are going for. What am I missing? :)

I didn't notice that - a mistake, lol. Should be in gitignore, lol.

File: data/share/bash-completion/completions/pycargoebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Gentoo Authors
```
Comment (mgorny) [inline data/share/bash-completion/completions/pycargoebuild:1]: Just put yourself here, "Gentoo Authors" outside Gentoo is pretty indeterminate.

File: data/share/bash-completion/completions/pycargoebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# Distributed under the terms of the GNU General Public License, v2 or later
```
Comment (mgorny) [inline data/share/bash-completion/completions/pycargoebuild:2]: …and use the SPDX thingy, for consistency.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,6 @@
 __pycache__/
 /.tox/
 /integration_test/dist/
+
+*.whl
+*.tar.gz
```
Comment (mgorny) [inline .gitignore:6]:
```suggestion
/dist/
```

?

**Review by mgorny — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**